pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -321,6 +321,16 @@ Iterable<ArchiveIssue> forbidGitDependencies(Pubspec pubspec) sync* {
   }
 }
 
+Iterable<ArchiveIssue> checkValidJson(String pubspecContent) sync* {
+  try {
+    final map = loadYaml(pubspecContent) as Map;
+    json.decode(json.encode(map)) as Map<String, dynamic>;
+  } on JsonUnsupportedObjectError catch (_) {
+    yield ArchiveIssue(
+        'pubspec.yaml contains values that can\'t be converted to JSON.');
+  }
```

Comment:
Actually, I like catching `JsonUnsupportedObjectError`.

But having a ` } catch (e) {` block which catches everything might also be nice...
We can make the two cache blocks do the same, but have the catch-all block print a warning using `package:logging`.

Reply:
First, I will catch JsonUnsupportedObjectError, and if there's any other ill add it to the logging. does this work?

Reply:
> This is a bit of a degenerate way of doing it... let's maybe catch any error :D
> 
> Also it might be interesting to write a one-off script and run this against all packages / all pubspec.yaml on pub.dev
> 
> * Packages names are available here: https://pub.dev/api/package-names
> * Find versions and tarballs for a package here: https://github.com/dart-lang/pub/blob/master/doc/repository-spec-v2.md

is there a constraint on using a specific language?

Reply:
@jonasfj I Wrote this script to download packages and validate them [script](https://gist.github.com/ramyak-mehra/802d60811914dd0b6472b4e11880d564) 
I think I  should use isolates to speed up the process but I am not sure how should I implement them. Possible implementations I can think of : 

- The isolate downloads the tar.gz files. Extract the pubspec content and returns it. Then the main thread validates the yaml and logs the entry somewhere.
- Any other ? happy to know.
- Completely ditch the isolates?

Reply:
Let's also add this script into this PR!
It could be under `pkg/pub_package_reader/tool/verify_pub_dev.dart`.